x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/percentiles_agg/config.ts

```diff
@@ -6,42 +6,38 @@
  */
 
 import { PercentilesAggForm } from './percentiles_form_component';
-import type { IPivotAggsConfigPercentiles } from './types';
+import type { IPivotAggsConfigPercentiles, PercentilesAggConfig, ValidationResult } from './types';
 import type { PivotAggsConfigBase } from '../../../../../../common';
 import {
   isPivotAggsConfigWithUiBase,
   PERCENTILES_AGG_DEFAULT_PERCENTS,
 } from '../../../../../../common';
 import type { PivotAggsConfigWithUiBase } from '../../../../../../common/pivot_aggs';
 
-/**
- * TODO this callback has been moved.
- * The logic of parsing the string should be improved.
- */
-function parsePercentsInput(inputValue: string | undefined) {
-  if (inputValue !== undefined) {
-    const strVals: string[] = inputValue.split(',');
-    const percents: number[] = [];
-    for (const str of strVals) {
-      if (str.trim().length > 0 && isNaN(str as any) === false) {
-        const val = Number(str);
-        if (val >= 0 && val <= 100) {
-          percents.push(val);
-        } else {
-          return [];
-        }
-      }
-    }
-
-    return percents;
+function validatePercentsInput(config: Partial<PercentilesAggConfig>): ValidationResult {
```

Comment:
Can we disable the Apply button if no percents have been entered? Currently you can apply with empty percents, and then get an error at the preview step.

Reply:
Done in: [#61fe474](https://github.com/elastic/kibana/pull/197816/commits/61fe47424eebbc8f674ed3ccb76c6b35403792d5)

Reply:
Regarding sorting – it's definitely possible, but what is the expected result? Should the values remain sorted when the user reopens the percentiles form, or should the columns in the results preview also be sorted?

Reply:
The idea was that we'd sort the percentiles presses the Apply button. But looks like the sorting in the preview table is handled differently, so that even if we sorted the percentiles in the popover, it wouldn't change the order in the table. Happy to leave this one for now.